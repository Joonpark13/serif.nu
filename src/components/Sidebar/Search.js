import React from 'react';
import PropTypes from 'prop-types';
import SearchBoxContainer from './SearchBoxContainer';
import SearchResultsContainer from './SearchResultsContainer';
import SectionSelectionContainer from './SectionSelectionContainer';
import AssociatedClassesSelectionContainer from './AssociatedClassesSelectionContainer';

const Search = ({ view }) => (
  <div>
    {view === 'search' && (
      <React.Fragment>
        <SearchBoxContainer />
        <SearchResultsContainer />
      </React.Fragment>
    )}
    {view === 'sectionSelection' && <SectionSelectionContainer />}
    {view === 'associatedClassesSelection' && <AssociatedClassesSelectionContainer />}
  </div>
);

Search.propTypes = {
  view: PropTypes.string.isRequired,
};

export default Search;
