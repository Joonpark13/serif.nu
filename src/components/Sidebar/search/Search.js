import React from 'react';
import PropTypes from 'prop-types';
import SearchBoxContainer from './SearchBoxContainer';
import SearchResultsContainer from './SearchResultsContainer';
import SearchSectionSelectionContainer from './SearchSectionSelectionContainer';
import SearchAssociatedClassesSelectionContainer from './SearchAssociatedClassesSelectionContainer';

const Search = ({ view }) => (
  <div>
    {view === 'search' && (
      <React.Fragment>
        <SearchBoxContainer />
        <SearchResultsContainer />
      </React.Fragment>
    )}
    {view === 'sectionSelection' && <SearchSectionSelectionContainer />}
    {view === 'associatedClassesSelection' && <SearchAssociatedClassesSelectionContainer />}
  </div>
);

Search.propTypes = {
  view: PropTypes.string.isRequired,
};

export default Search;
