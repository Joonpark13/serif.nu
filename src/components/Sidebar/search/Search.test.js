import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Search view="search" />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('renders sectionSelection view correctly', () => {
    const wrapper = shallow(<Search view="sectionSelection" />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('renders associatedClassesSelection view correctly', () => {
    const wrapper = shallow(<Search view="associatedClassesSelection" />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
