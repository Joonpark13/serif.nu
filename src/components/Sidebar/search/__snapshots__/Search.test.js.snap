// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders associatedClassesSelection view correctly 1`] = `
<div>
  <Connect(Component) />
</div>
`;

exports[`Search renders correctly 1`] = `
<div>
  <React.Fragment>
    <Connect(WithStyles(SearchBox)) />
    <Connect(Component) />
  </React.Fragment>
</div>
`;

exports[`Search renders sectionSelection view correctly 1`] = `
<div>
  <Connect(Component) />
</div>
`;
